cmake/modules/RootMacros.cmake

```diff
@@ -973,8 +973,15 @@ function(ROOT_LINKER_LIBRARY library)
     endforeach()
   endif()
 
+  if(rpath)
+    ROOT_APPEND_LIBDIR_TO_BUILD_RPATH(${library})
+  endif()
```

Comment:
Hopefully, CMake does this already. Can we try running without?

>  By default, CMake sets the runtime path of binaries in the build tree to contain search paths it knows are needed to find the shared libraries they link. Projects may set BUILD_RPATH to specify additional search paths.

from https://cmake.org/cmake/help/latest/prop_tgt/BUILD_RPATH.html